initrd/bin/kexec-save-default

```diff
@@ -129,6 +129,41 @@ if [ ! -r $ENTRY_FILE -o ! -r $HASH_FILE ]; then
 	die "Failed to write default config"
 fi
 
+# logic to parse OS initrd to get extract crypttab, its filepaths and its OS defined options
+mkdir -p /tmp/initrd_extract
+cd /tmp/initrd_extract
+# User just selected a default boot option. Get initrd filename
+current_default_initrd=$(cat /boot/kexec_default_hashes.txt | grep initr | awk -F " " {'print $NF'} | sed 's/\.\//\/boot\//g')
+# Get crypttab files paths from initrd
+crypttab_files=$(zcat $current_default_initrd | cpio --list --quiet | grep crypttab) > /dev/null || crypttab_files=$(cpio --list --quiet < $current_default_initrd | grep crypttab) > /dev/null
+
+if [ ! -z "$crypttab_files" ]; then
+	echo "Extracting current selected default boot's $current_default_initrd for crypttab analysis..."
+	zcat $current_default_initrd | cpio -id --quiet "$crypttab_files" > /dev/null || cpio -id --quiet < $current_default_initrd "$crypttab_files" > /dev/null
+
+	crypttab_entries=$(cat "$crypttab_files"|uniq| sed 's/none/\/secret.key/g')
+	cd - > /dev/null
+
+	#insert current default boot's initrd crypttab location into tracking file to be overwritten into initramfs
+	echo "The following OS crypttab was found in intrd and Heads will overwrite it: $crypttab_files"
+	echo "Adding found crypptab files location under $bootdir/kexec_initrd_crypttab_path.txt"	
```

Comment:
I don't think this is important or useful; it's always `kexec_initrd_crypttab_path.txt` under the boot dir.  IMO, I think we could remove it so the relevant info is more prominent.

I do like the other two traces though, both of those are characteristics of the initrd we found that would be important for troubleshooting.

Reply:
@JonathonHall-Purism : Please consider this (as general guideline for user understanding of the minimal technicalities to underatand and therefore trust the process):

- We add two new /boot/kexec_* files.
- We do so otherwise in an obscure place (kexec-save-boot)
  - The actual reason we are doing it there is because this is the place where whatever is modifying /boot gets prompted to be signed without the checksum/dirtree validation to kick in detecting the changes.

Therefore, unless you keep your stance here, I would change the behavior everywhere else in heads to tell the user that some /boot/kexec_* files were added/modified instead of respecting actual Heads behavior which lacks output for user to understand what is happening.

But if you keep your stance here, I have no **strong** opinion. I only remember my own learning path of Heads, and those lines are actually really good entry point to inspect code change and where things are happening.
@JonathonHall-Purism tell me! Otherwise I would really love to have a quick **HEADS_DEBUG=y** possible in board configs and the new code changes implementing those for additional output when developing new features.

Reply:
Yeah, I understand what you're saying, and I do agree.  We can keep it :+1:   Definitely does help newcomers that want to understand how it works and hack on it, anybody that does look into their own /boot, etc.  Not really a strong opinion on my end either, let's keep it.